grid2op/Opponent/NeuripsOpponent.py

```diff
@@ -0,0 +1,136 @@
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+import warnings
+import numpy as np
+
+from grid2op.Opponent import BaseOpponent
+from grid2op.Exceptions import OpponentError
+
+
+class NeuripsOpponent(BaseOpponent):
+    def __init__(self, action_space):
+        BaseOpponent.__init__(self, action_space)
+        self._do_nothing = None
+        self._attacks = None
+        self._lines_ids = None
+        self._next_attack_time = None
+        self._attack_period = None
+        self._rho_normalization = None
+
+        # this is the constructor:
+        # it should have the exact same signature as here
+
+    def init(self, lines_attacked=[], rho_normalization=[], **kwargs):
+        # this if the function used to properly set the object.
+        # It has the generic signature above,
+        # and it's way more flexible that the other one.
+
+        if len(lines_attacked) == 0:
+            warnings.warn(f'The opponent is deactivated as there is no information as to which line to attack. '
+                          f'You can set the argument "kwargs_opponent" to the list of the line names you want '
+                          f' the opponent to attack in the "make" function.')
+
+        # Store attackable lines IDs
+        self._lines_ids = []
+        for l_name in lines_attacked:
+            l_id = np.where(self.action_space.name_line == l_name)
+            if len(l_id) and len(l_id[0]):
+                self._lines_ids.append(l_id[0][0])
+            else:
+                raise OpponentError("Unable to find the powerline named \"{}\" on the grid. For "
+                                    "information, powerlines on the grid are : {}"
+                                    "".format(l_name, sorted(self.action_space.name_line)))
+
+        # Pre-build attacks actions
+        self._do_nothing = self.action_space({})
+        self._attacks = []
+        for l_id in self._lines_ids:
+            a = self.action_space({
+                'set_line_status': [(l_id, -1)]
+            })
+            self._attacks.append(a)
+        self._attacks = np.array(self._attacks)
+
+        # Usage rates normalization
+        self._rho_normalization = np.ones_like(lines_attacked)
+        if len(rho_normalization) == 0:
+            warnings.warn('The usage rate normalization is not specified. No normalization will be performed.')
+        elif len(rho_normalization) != len(lines_attacked):
+            warnings.warn(f'The usage rate normalization must have the same length as the number '
+                          f'of attacked lines. No normalization will be performed.')
+        else:
+            self._rho_normalization = np.array(rho_normalization)
+
+        # Opponent's attack period
+        self._attack_period = kwargs["attack_period"] if "attack_period" in kwargs else 12*24
```

Comment:
then directly expand the "attack_period" to be a keyword arguments: `def init(self, lines_attacked=[], rho_normalization=[], attack_period = 12*24**kwargs):`

Reply:
done